src/main/java/sirius/kernel/commons/Limit.java

```diff
@@ -135,6 +135,17 @@ public int getTotalItems() {
         return maxItems == 0 ? 0 : itemsToSkip + maxItems;
     }
 
+    /**
+     * Returns the remaining number of items to be processed. This is the number of items to be skipped plus the
+     * number of items to be accepted minus the number of items already processed.
+     * Returns <tt>null</tt> to indicate that there is no upper limit.
+     *
+     * @return the remaining number of items to be processed or <tt>null</tt> if there is no upper limit
+     */
+    public Integer getRemainingItems() {
```

Comment:
int and default to 0 ?

Reply:
in contrast to getTotalItems, if enough items are processed, this method can actually return 0

so if it was changed to int returning 0 could mean either

a) unlimited items
b) all items already processed 

thats why i think returning null for the unlimited case is better